src/main/java/org/jboss/modules/Main.java

```diff
@@ -520,11 +524,16 @@ public static void main(String[] args) throws Throwable {
         final String logManagerName = getServiceName(bootClassLoader, "java.util.logging.LogManager");
         if (logManagerName != null) {
             System.setProperty("java.util.logging.manager", logManagerName);
+            // Special handling is required for cases when a log manager is on the boot class path
+            configureLogManagerModuleFinder(logManagerName);
             if (LogManager.getLogManager().getClass() == LogManager.class) {
                 System.err.println("WARNING: Failed to load the specified log manager class " + logManagerName);
             } else {
                 Module.setModuleLogger(new JDKModuleLogger());
             }
+        } else {
+            // Special handling is required for cases when a log manager is on the boot class path
+            configureLogManagerModuleFinder(System.getProperty("java.util.logging.manager"));
```

Comment:
I'm guessing this is an unlikely scenario and maybe we shouldn't do this, but I've kept it for now. Let me know if we should remove this part.

Reply:
TBH I'd go way simpler, and avoid reflection (which can come back to bite later).  Just add a block of code in a method like this:

```java
try {
    LogModuleResolverHolder.setResolver(...);
} catch (LinkageError ignored) {
}
```

Then make sure JBoss Modules is compiled against LogManager as a provided, optional dep.  Since JBoss LogManager depends on JBoss Modules, you could break the cycle by putting the couple of classes in a tiny third project, or by using the dependency plugin to fetch it in and add it to compilation as an `additionalClassPathElement`, or just add a `stubs` where a stubbed-out version of that class is included instead of depending on LogManager (you need a separate compiler plugin execution to compile the stubs and again use `additionalClassPathElement` to make sure they're visible to the compilation phase of Modules).

OK not really simpler from a Maven POV but from a code POV it's, like, a couple lines only!

Reply:
A small third project may make sense. Just a simple `ModuleFinder` project which any project could use as this is likely a common pattern for Java 8/Java 11 projects that need something special.

That said that's another JAR that will be required on the boot class path and the system packages.

Reply:
I guess https://github.com/jboss-modules/jboss-modules/pull/244 is a follow up on this discussion, correct guys?

Reply:
@ropalka yes. I don't think there has been a determination on how we want this done. So far this is the only way I've got it to work.